app/components/Poll/index.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import React from 'react';
+import Button from 'app/components/Button';
+import styles from './Poll.css';
+import type { PollEntity, OptionEntity } from 'app/reducers/polls';
+import { Link } from 'react-router';
+import Icon from 'app/components/Icon';
+import { Flex } from 'app/components/Layout';
+
+type Props = {
+  poll: PollEntity,
+  handleVote: (pollId: number, optionId: number) => Promise<*>,
+  backgroundLight?: boolean,
+  truncate?: number,
+  details?: boolean
+};
+
+type State = {
+  truncateOptions: boolean,
+  optionsToShow: Array<OptionEntity>,
+  expanded: boolean
+};
+
+class Poll extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { options } = props.poll;
+    if (props.truncate && options.length > props.truncate) {
+      this.state = {
+        truncateOptions: true,
+        optionsToShow: options.slice(0, 3),
+        expanded: false
+      };
+    } else if (!props.truncate) {
+      this.state = {
+        truncateOptions: false,
+        optionsToShow: options,
+        expanded: true
+      };
+    } else {
+      this.state = {
+        truncateOptions: false,
+        optionsToShow: options,
+        expanded: true
+      };
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.poll.options !== this.props.poll.options) {
+      this.props.truncate && !this.state.expanded
+        ? this.setState({
+            optionsToShow: this.props.poll.options.slice(0, this.props.truncate)
+          })
+        : this.setState({ optionsToShow: this.props.poll.options });
+    }
+  }
+
+  render() {
+    const { poll, handleVote, backgroundLight, details, truncate } = this.props;
+    const { truncateOptions, optionsToShow, expanded } = this.state;
+    const { id, title, description, options, hasAnswered, totalVotes } = poll;
+
+    const toggleTruncate = () =>
```

Comment:
Maybe make this a function/method inside the class, instead of having it inside the render func?